parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -330,36 +354,55 @@ public static Binary fromByteArray(final byte[] value) {
   }
 
   private static class ByteBufferBackedBinary extends Binary {
```

Comment:
We could simplify significantly by having the ByteBuffer based impl the only one.
bytes can always be wrapped and ByteBuffer.hasArray can be used to return it with no copy if this is the case.